app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -24,14 +28,33 @@ void setUp() {
 
     @Test
     void shouldAllowSequentialConnections() throws IOException {
+        int numberOfThreadsToGenerate = 1;
         String[] clientRequests = {"curl foo1", "curl foo2", "curl foo3", "curl foo4"};
         ReactorMock reactor = new ReactorMock(clientRequests);
 
         new Communicator(
+                new ThreadGeneratorMock(numberOfThreadsToGenerate),
                 new ListenerMock(new String[]{"foo", clientHasDisconnected}),
                 new MessengerMock()
         ).communicate(reactor);
 
         assertEquals(clientRequests.length, reactor.numberOfAcceptedClients);
     }
+
+    @Property
+    void shouldAllowMultipleConnections(@ForAll @IntRange(min = 1, max = 10) Integer numberOfThreadsToGenerate) throws IOException {
```

Comment:
How is this test any different from the one above? How do we discern between sequential connections and multiple connections? For example, looking at the setup, the way client requests are made is the same but in reality, it's very different. Maybe this test should be `create as many threads as there are connections`?

Reply:
Hmmm, I think I see what you mean, the name is not necessarily accurate/properly descriptive 🤔, @mrfishball are you available for a quick pair to discuss this?

Reply:
Not only that, I think what we really want to test is that it spins up 2 threads if there are 2 connections/requests.